label_studio/storage/filesystem.py

```diff
@@ -183,7 +184,7 @@ def readable_path(self):
         return self.path
 
     def _get_value(self, key):
-        return self.data[int(key)]
+        return deepcopy(self.data[int(key)])
```

Comment:
Maybe it's better to put it right in `resolve_task_data_uri()`? I'm concerned about this unnecessary step in the case when URLs should not be resolved

Reply:
That was my initial though also. Then I thought that ExternalTasksJSONStorage is used by default and it is semantically bit different compared to other storages which will fetch a copy of the data always. Putting it there would prevent similar issues being introduced later and task is currently fairly light weight object.

That said, either way will fix the issue I'm facing, so I'm definitely fine with putting it to resolve_task_data_uri() also.

Reply:
Can I merge it? @niklub

Reply:
@hepaajan I agree, guess it's better to unify storage design to avoid further issues, even at the cost of some minor performance reduction. @makseq let's merge it with https://github.com/heartexlabs/label-studio/pull/350